crates/subspace-farmer/src/bin/subspace-farmer/main.rs

```diff
@@ -159,20 +260,29 @@ async fn main() -> Result<()> {
 
     match command.subcommand {
         Subcommand::Wipe => {
-            commands::wipe(&base_path)?;
+            if command.farm.is_empty() {
+                commands::wipe(&base_path)?;
+            } else {
+                unimplemented!()
+            }
+
             info!("Done");
         }
         Subcommand::Farm(farming_args) => {
-            if !base_path.exists() {
-                fs::create_dir_all(&base_path).unwrap_or_else(|error| {
-                    panic!(
-                        "Failed to create data directory {:?}: {:?}",
-                        base_path, error
-                    )
-                });
-            }
+            if command.farm.is_empty() {
+                if !base_path.exists() {
+                    fs::create_dir_all(&base_path).unwrap_or_else(|error| {
+                        panic!(
+                            "Failed to create data directory {:?}: {:?}",
+                            base_path, error
+                        )
+                    });
```

Comment:
:thinking: maybe it would be better to propagate error like [this](https://docs.rs/anyhow/latest/anyhow/trait.Context.html#tymethod.with_context). We could have nice error reporter which unwraps errors and sources one by one and shows user detailed info

Reply:
I think such refactoring (there are other places already) is outside of this PR either way